core/src/storage2/lazy/mod.rs

```diff
@@ -44,7 +44,12 @@ pub use self::{
         LazyArrayLength,
     },
     lazy_cell::LazyCell,
-    lazy_hmap::LazyHashMap,
+    lazy_hmap::{
+        Entry,
+        LazyHashMap,
+        OccupiedEntry,
+        VacantEntry,
+    },
```

Comment:
You should not expose the `Entry`, `OccupiedEntry` and `VacantEntry` to the `lazy` module and instead should make the `lazy_hmap` module simply public. This way we would not have needed the `Entry` -> `InternalEntry` rename. The std lib does the same for those utility types. So you have an `Entry` per map collection there as well.
However, I agree that to counteract confusion we should rename `Entry` to something like `StorageEntry` instead of `InternalEntry` because that describes better what it is.